目标平台为 MinGW 时，Clang 工具链参数 `--target=x86_64-windows-gnu` 不适用于 LLVM-MinGW

### Xmake 版本

xmake v2.9.6+HEAD.abd15c255

### 操作系统版本和架构

Windows 10 / Arch Linux

### 描述问题

运行 `xrepo install -vD -p mingw --toolchain=clang gtest` 报错
```
fatal error: 'algorithm' file not found
  137 | #include <algorithm>
      |          ^~~~~~~~~~~
```
在诊断日志中可以看到，Clang 接受了 `--target=x86_64-windows-gnu` 参数，这一参数可能并不适用于 `LLVM-MinGW`

运行 `clang -v --target=x86_64-windows-gnu src\hello.c` 编译一个简单的 C 程序，会产生链接错误

`LLVM-MinGW` 将 标准库 放置在`x86_64-w64-mingw32` , `aarch64-w64-mingw32` 等目录下，目录中包含 `bin/libc++.dll` , `lib/crt1.o` 等文件。使用 `clang -v --target=x86_64-w64-mingw32 src\hello.c` 可以正常编译。

在 Arch 上测试，发现 `--target=x86_64-windows-gnu` 与 `--target=x86_64-w64-mingw32` 都可以正常编译。 Arch 的 MinGW 标准库的路径是 `/usr/x86_64-w64-mingw32`。

因此，目标平台为 MinGW 时，Clang 的 target 设定为 `x86_64-w64-mingw32`，或 `x86_64-w64-windows-gnu` 可能更好

### 期待的结果

Clang 能够正常地进行跨平台编译

### 工程配置

基于 XMake 的包 ：`xrepo install -vD -p mingw --toolchain=clang -f "main=true" gtest`
基于 CMake 的包 ：`xrepo install -vD -p mingw --toolchain=clang gflags`

### 附加信息和错误日志

`xrepo install -vD -p mingw --toolchain=clang -f "main=true" gtest`
---
```
[ 50%]: cache compiling.release googletest\src\gtest-all.cc
clang -c -Qunused-arguments -m64 --target=x86_64-windows-gnu -std=c++14 -Igoogletest\include -Igoogletest -o build_4caca900\.objs\gtest\mingw\x86_64\release\googletest\src\gtest-all.cc.obj googletest\src\gtest-all.cc
error: @programdir\core\main.lua:329: @programdir\actions\build\main.lua:148: @programdir\modules\async\runjobs.lua:325: @programdir\modules\private\action\build\object.lua:91: @programdir\modules\core\tools\gcc.lua:966: In file included from googlemock\src\gmock-all.cc:39:
In file included from googlemock\include/gmock/gmock.h:56:
googlemock\include/gmock/gmock-actions.h:137:10: fatal error: 'algorithm' file not found
  137 | #include <algorithm>
      |          ^~~~~~~~~~~
1 error generated.
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\modules\core\tools\gcc.lua:966]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [@programdir\modules\core\tools\gcc.lua:907]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]:
    [@programdir\core\tool\compiler.lua:288]: in function 'compile'
    [@programdir\modules\private\action\build\object.lua:91]: in function 'script'
    [@programdir\modules\private\action\build\object.lua:122]: in function 'build_object'
    [@programdir\modules\private\action\build\object.lua:147]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:117]: in function 'try'
    [@programdir\modules\async\runjobs.lua:223]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:406]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:1004: in function 'base/os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:329: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
error: @programdir\core\sandbox\modules\os.lua:378: execv(D:\Applications\Scoop\apps\xmake\current\xmake.exe build --diagnosis --verbose --jobs=14) failed(-1)
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\core\sandbox\modules\os.lua:378]:
    [@programdir\core\sandbox\modules\os.lua:291]: in function 'vrunv'
    [@programdir\modules\package\tools\xmake.lua:531]: in function 'install'
    [D:\projects\xmake-repo\packages\g\gtest\xmake.lua:56]: in function 'script'
    [...dir\modules\private\action\require\impl\utils\filter.lua:114]: in function 'call'
    [...\modules\private\action\require\impl\actions\install.lua:452]:

  => install gtest v1.15.2 .. failed
error: @programdir\core\main.lua:329: @programdir\modules\async\runjobs.lua:325: ...\modules\private\action\require\impl\actions\install.lua:561: install failed!
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [...\modules\private\action\require\impl\actions\install.lua:561]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [...\modules\private\action\require\impl\actions\install.lua:418]:
    [...modules\private\action\require\impl\install_packages.lua:510]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:1004: in function 'os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:329: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
error: @programdir\core\main.lua:329: @programdir\core\sandbox\modules\os.lua:378: execv(D:\Applications\Scoop\apps\xmake\current\xmake.exe require -v -D -j 14 --extra={system=false,configs={main=true}} gtest) failed(-1)
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\core\sandbox\modules\os.lua:378]:
    [@programdir\modules\private\xrepo\action\install.lua:309]: in function '_install_packages'
    [@programdir\modules\private\xrepo\action\install.lua:316]:
    [@programdir\modules\private\xrepo\main.lua:196]:
    [@programdir\plugins\lua\main.lua:124]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]:
    [@programdir\core\base\task.lua:491]: in function 'run'
    [@programdir\core\main.lua:327]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:406]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:1004: in function 'os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:329: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
```

`xrepo install -vD -p mingw --toolchain=clang gflags`
---
```
cmake -DBUILD_TESTING=OFF -DGFLAGS_REGISTER_BUILD_DIR=OFF -DGFLAGS_REGISTER_INSTALL_PREFIX=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DBUILD_STATIC_LIBS=ON -DBUILD_gflags_LIB=ON -DBUILD_gflags_nothreads_LIB=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_INSTALL_PREFIX=C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\6758891cfbde4f4cb92634e6368a5498 -DCMAKE_INSTALL_LIBDIR:PATH=lib -G Ninja "-DCMAKE_CXX_FLAGS=-m64 --target=x86_64-windows-gnu" -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=BOTH -DCMAKE_C_COMPILER=clang.exe "-DCMAKE_ASM_FLAGS=-m64 --target=x86_64-windows-gnu" "-DCMAKE_SHARED_LINKER_FLAGS=-m64 --target=x86_64-windows-gnu" "-DCMAKE_EXE_LINKER_FLAGS=-m64 --target=x86_64-windows-gnu" -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_ROOT_PATH=D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/ -DCMAKE_STATIC_LINKER_FLAGS= -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=BOTH -DCMAKE_SYSTEM_PROCESSOR=x86_64 -DCMAKE_MAKE_PROGRAM=ninja -DHAVE_FLAG_SEARCH_PATHS_FIRST=0 -DCMAKE_ASM_COMPILER=clang.exe -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=BOTH "-DCMAKE_MODULE_LINKER_FLAGS=-m64 --target=x86_64-windows-gnu" -DCMAKE_RC_COMPILER=llvm-rc.exe -DCMAKE_CXX_COMPILER=clang++.exe "-DCMAKE_C_FLAGS=-m64 --target=x86_64-windows-gnu" -DCMAKE_AR=ar.exe -DCMAKE_RANLIB=ranlib.exe C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source
CMake Deprecation Warning at CMakeLists.txt:73 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The CXX compiler identification is Clang 19.1.5
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/bin/clang++.exe
-- Check for working CXX compiler: D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/bin/clang++.exe - broken
CMake Error at D:/Applications/Scoop/apps/cmake/3.31.1/share/cmake-3.31/Modules/CMakeTestCXXCompiler.cmake:73 (message):
  The C++ compiler

    "D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/bin/clang++.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: 'C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c/CMakeFiles/CMakeScratch/TryCompile-0ba15h'

    Run Build Command(s): ninja -v cmTC_dc6f8
    [1/2] D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\clang++.exe   -m64 --target=x86_64-windows-gnu -MD -MT CMakeFiles/cmTC_dc6f8.dir/testCXXCompiler.cxx.obj -MF CMakeFiles\cmTC_dc6f8.dir\testCXXCompiler.cxx.obj.d -o CMakeFiles/cmTC_dc6f8.dir/testCXXCompiler.cxx.obj -c C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c/CMakeFiles/CMakeScratch/TryCompile-0ba15h/testCXXCompiler.cxx
    [2/2] C:\Windows\system32\cmd.exe /C "cd . && D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\clang++.exe -m64 --target=x86_64-windows-gnu  CMakeFiles/cmTC_dc6f8.dir/testCXXCompiler.cxx.obj -o cmTC_dc6f8.exe -Wl,--out-implib,libcmTC_dc6f8.dll.a -Wl,--major-image-version,0,--minor-image-version,0  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && cd ."
    FAILED: cmTC_dc6f8.exe
    C:\Windows\system32\cmd.exe /C "cd . && D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\clang++.exe -m64 --target=x86_64-windows-gnu  CMakeFiles/cmTC_dc6f8.dir/testCXXCompiler.cxx.obj -o cmTC_dc6f8.exe -Wl,--out-implib,libcmTC_dc6f8.dll.a -Wl,--major-image-version,0,--minor-image-version,0  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && cd ."
    clang-19: error: unable to execute command: Couldn't execute program 'D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/19.1.5-20241203/bin/ld': Unknown error (0xC1)
    clang-19: error: linker command failed with exit code 1 (use -v to see invocation)
    ninja: build stopped: subcommand failed.





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:97 (project)


-- Configuring incomplete, errors occurred!
error: @programdir\core\sandbox\modules\os.lua:378: execv(cmake -DBUILD_TESTING=OFF -DGFLAGS_REGISTER_BUILD_DIR=OFF -DGFLAGS_REGISTER_INSTALL_PREFIX=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DBUILD_STATIC_LIBS=ON -DBUILD_gflags_LIB=ON -DBUILD_gflags_nothreads_LIB=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_INSTALL_PREFIX=C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\6758891cfbde4f4cb92634e6368a5498 -DCMAKE_INSTALL_LIBDIR:PATH=lib -G Ninja "-DCMAKE_CXX_FLAGS=-m64 --target=x86_64-windows-gnu" -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=BOTH -DCMAKE_C_COMPILER=clang.exe "-DCMAKE_ASM_FLAGS=-m64 --target=x86_64-windows-gnu" "-DCMAKE_SHARED_LINKER_FLAGS=-m64 --target=x86_64-windows-gnu" "-DCMAKE_EXE_LINKER_FLAGS=-m64 --target=x86_64-windows-gnu" -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_ROOT_PATH=D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/ -DCMAKE_STATIC_LINKER_FLAGS= -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=BOTH -DCMAKE_SYSTEM_PROCESSOR=x86_64 -DCMAKE_MAKE_PROGRAM=ninja -DHAVE_FLAG_SEARCH_PATHS_FIRST=0 -DCMAKE_ASM_COMPILER=clang.exe -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=BOTH "-DCMAKE_MODULE_LINKER_FLAGS=-m64 --target=x86_64-windows-gnu" -DCMAKE_RC_COMPILER=llvm-rc.exe -DCMAKE_CXX_COMPILER=clang++.exe "-DCMAKE_C_FLAGS=-m64 --target=x86_64-windows-gnu" -DCMAKE_AR=ar.exe -DCMAKE_RANLIB=ranlib.exe C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source) failed(1)
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\core\sandbox\modules\os.lua:378]:
    [@programdir\core\sandbox\modules\os.lua:291]: in function 'vrunv'
    [@programdir\modules\package\tools\cmake.lua:1317]: in function 'configure'
    [@programdir\modules\package\tools\cmake.lua:1359]: in function 'install'
    [D:\projects\xmake-repo\packages\g\gflags\xmake.lua:44]: in function 'script'
    [...dir\modules\private\action\require\impl\utils\filter.lua:114]: in function 'call'
    [...\modules\private\action\require\impl\actions\install.lua:452]:

  => install gflags v2.2.2 .. failed
error: @programdir\core\main.lua:329: @programdir\modules\async\runjobs.lua:325: ...\modules\private\action\require\impl\actions\install.lua:561: install failed!
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [...\modules\private\action\require\impl\actions\install.lua:561]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [...\modules\private\action\require\impl\actions\install.lua:418]:
    [...modules\private\action\require\impl\install_packages.lua:510]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:1004: in function 'base/os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:329: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
error: @programdir\core\main.lua:329: @programdir\core\sandbox\modules\os.lua:378: execv(D:\Applications\Scoop\apps\xmake\current\xmake.exe require -v -D -j 14 --extra={system=false} gflags) failed(-1)
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\core\sandbox\modules\os.lua:378]:
    [@programdir\modules\private\xrepo\action\install.lua:309]: in function '_install_packages'
    [@programdir\modules\private\xrepo\action\install.lua:316]:
    [@programdir\modules\private\xrepo\main.lua:196]:
    [@programdir\plugins\lua\main.lua:124]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]:
    [@programdir\core\base\task.lua:491]: in function 'run'
    [@programdir\core\main.lua:327]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:406]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:1004: in function 'os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:329: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
```

目标为 Windows ，设置工具链为 Clang，基于 XMake 的包会使用 Clang，基于CMake 的包仍尝试用 MSVC 安装，安装完成后的测试用 Clang，有点怪。

<details>
<summary>基于 XMake 的包</summary>

```
$ xrepo install -vD --toolchain=clang gtest
D:\projects\xmake\core\build\xmake.exe f -c --require=n -vD --toolchain=clang
checking for platform ... windows
checking for architecture ... x64
checking for clang ... ok
checking for clang++ ... ok
checking for the linker (ld) ... clang++
checking for clang++ ... ok
checking for the shared library linker (sh) ... clang++
configure
{
    toolchain = clang
    clean = true
    mingw = D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/
    buildir = build
    proxy = socks5://127.0.0.1:20170
    proxy_pac = pac.lua
    mode = release
    theme = default
    network = public
    arch = x64
    plat = windows
    host = windows
    ccache = true
    kind = static
    ndk_stdcxx = true
}
D:\projects\xmake\core\build\xmake.exe require -v -D -j 14 --extra={system=false} gtest
checking for git ... ok
checking for gzip ... ok
checking for tar ... ok
git rev-parse HEAD
finding gtest from xmake ..
checking for xmake::gtest ... no
note: install or modify (m) these packages (pass -y to skip confirm)?
in myrepo:
  -> gtest v1.15.2 [runtimes:"MT", toolchains:"clang", license:BSD-3]
please input: y (y/n/m)

checking for ping ... ok
pinging the host(github.com) ... 105 ms
checking for 7z ... ok
7z x -y v1.15.2.zip -snl- -osource.tmp

7-Zip 24.09 (x64) : Copyright (c) 1999-2024 Igor Pavlov : 2024-11-29

Scanning the drive for archives:
1 file, 1097779 bytes (1073 KiB)

Extracting archive: v1.15.2.zip
--
Path = v1.15.2.zip
Type = zip
Physical Size = 1097779
Comment = b514bdc898e2951020cbdca1304b75f5950d1f59

Everything is Ok

Folders: 30
Files: 247
Size:       4015306
Compressed: 1097779
D:\projects\xmake\core\build\xmake.exe f --diagnosis --verbose -y -c --plat=windows --arch=x64 --mode=release --kind=static --runtimes=MT --buildir=build_472ce5b1
checking for cl.exe ... C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\bin\HostX64\x64\cl.exe
checking for Microsoft Visual Studio (x64) version ... 2022
checking for Microsoft C/C++ Compiler (x64) version ... 19.42.34435
checkinfo: cannot runv(nim.exe --version), No such file or directory
checking for nim ... no
checkinfo: cannot runv(nim.exe --version), No such file or directory
checking for nim ... no
checking for clang ... ok
checking for clang ... ok
checking for the c++ compiler (cxx) ... clang
configure
{
    network = public
    host = windows
    buildir = build_472ce5b1
    vs = 2022
    runtimes = MT
    mingw = D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/
    mode = release
    clean = true
    kind = static
    theme = default
    plat = windows
    arch = x64
    proxy = socks5://127.0.0.1:20170
    ndk_stdcxx = true
    proxy_pac = pac.lua
    ccache = true
}
D:\projects\xmake\core\build\xmake.exe build --diagnosis --verbose --jobs=14
checking for flags (clang_ms_runtime_lib) ... ok
> clang "-fms-runtime-lib=dll" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
[ 50%]: cache compiling.release googletest\src\gtest-all.cc
clang -c -Qunused-arguments -m64 --target=x86_64-windows-msvc -std=c++14 -fms-runtime-lib=static -Igoogletest\include -Igoogletest -fexceptions -fcxx-exceptions -o build_472ce5b1\.objs\gtest\windows\x64\release\googletest\src\gtest-all.cc.obj googletest\src\gtest-all.cc
checking for flags (-MMD -MF) ... ok
> clang "-MMD" "-MF" "C:\Users\leemu\AppData\Local\Temp\.xmake\241216\_4E03EF5466F648308CA736C8466C7690" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
checking for flags (-fdiagnostics-color=always) ... ok
> clang "-fdiagnostics-color=always" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
checking for flags (-fansi-escape-codes) ... ok
> clang "-fansi-escape-codes" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
checking for flags (-Wno-gnu-line-marker -Werror) ... ok
> clang "-Wno-gnu-line-marker" "-Werror" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
[ 50%]: cache compiling.release googlemock\src\gmock-all.cc
clang -c -Qunused-arguments -m64 --target=x86_64-windows-msvc -std=c++14 -fms-runtime-lib=static -Igooglemock\include -Igooglemock -Igoogletest\include -Igoogletest -fexceptions -fcxx-exceptions -o build_472ce5b1\.objs\gmock\windows\x64\release\googlemock\src\gmock-all.cc.obj googlemock\src\gmock-all.cc
error: @programdir\core\main.lua:329: @programdir\actions\build\main.lua:148: @programdir\modules\async\runjobs.lua:325: @programdir\modules\private\action\build\object.lua:91: @programdir\modules\core\tools\gcc.lua:966: In file included from googletest\src\gtest-all.cc:38:
googletest\include/gtest/gtest.h:52:10: fatal error: 'cstddef' file not found
   52 | #include <cstddef>
      |          ^~~~~~~~~
1 error generated.
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\modules\core\tools\gcc.lua:966]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [@programdir\modules\core\tools\gcc.lua:908]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]:
    [@programdir\core\tool\compiler.lua:288]: in function 'compile'
    [@programdir\modules\private\action\build\object.lua:91]: in function 'script'
    [@programdir\modules\private\action\build\object.lua:122]: in function 'build_object'
    [@programdir\modules\private\action\build\object.lua:147]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:117]: in function 'try'
    [@programdir\modules\async\runjobs.lua:224]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:406]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:1004: in function 'os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:329: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
error: @programdir\core\sandbox\modules\os.lua:378: execv(D:\projects\xmake\core\build\xmake.exe build --diagnosis --verbose --jobs=14) failed(-1)
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\core\sandbox\modules\os.lua:378]:
    [@programdir\core\sandbox\modules\os.lua:291]: in function 'vrunv'
    [@programdir\modules\package\tools\xmake.lua:531]: in function 'install'
    [D:\projects\xmake-repo\packages\g\gtest\xmake.lua:56]: in function 'script'
    [...dir\modules\private\action\require\impl\utils\filter.lua:114]: in function 'call'
    [...\modules\private\action\require\impl\actions\install.lua:452]:

  => install gtest v1.15.2 .. failed
error: @programdir\core\main.lua:329: @programdir\modules\async\runjobs.lua:325: ...\modules\private\action\require\impl\actions\install.lua:561: install failed!
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [...\modules\private\action\require\impl\actions\install.lua:561]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [...\modules\private\action\require\impl\actions\install.lua:419]:
    [...modules\private\action\require\impl\install_packages.lua:510]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:1004: in function 'os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:329: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
error: .\xmake\core\main.lua:329: .\xmake\core\sandbox\modules\os.lua:378: execv(D:\projects\xmake\core\build\xmake.exe require -v -D -j 14 --extra={system=false} gtest) failed(-1)
stack traceback:
    [C]: in function 'error'
    [.\xmake\core\base\os.lua:1004]:
    [.\xmake\core\sandbox\modules\os.lua:378]:
    [.\xmake\modules\private\xrepo\action\install.lua:309]: in function '_install_packages'
    [.\xmake\modules\private\xrepo\action\install.lua:316]:
    [.\xmake\modules\private\xrepo\main.lua:196]:
    [.\xmake\plugins\lua\main.lua:124]:
    [C]: in function 'xpcall'
    [.\xmake\core\base\utils.lua:244]:
    [.\xmake\core\base\task.lua:491]: in function 'run'
    [.\xmake\core\main.lua:327]: in function 'cotask'
    [.\xmake\core\base\scheduler.lua:406]:

stack traceback:
        [C]: in function 'error'
        .\xmake\core\base\os.lua:1004: in function 'base/os.raiselevel'
        (...tail calls...)
        .\xmake\core\main.lua:329: in upvalue 'cotask'
        .\xmake\core\base\scheduler.lua:406: in function <.\xmake\core\base\scheduler.lua:399>
```
</details>


<details>
<summary>基于 CMake 的包</summary>

```
$ xrepo install -vD --toolchain=clang gflags
D:\projects\xmake\core\build\xmake.exe f -c --require=n -vD --toolchain=clang
checking for platform ... windows
checking for architecture ... x64
checking for clang ... ok
checking for clang++ ... ok
checking for the linker (ld) ... clang++
checking for clang++ ... ok
checking for the shared library linker (sh) ... clang++
configure
{
    host = windows
    mingw = D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/
    clean = true
    theme = default
    toolchain = clang
    arch = x64
    buildir = build
    ccache = true
    ndk_stdcxx = true
    proxy = socks5://127.0.0.1:20170
    kind = static
    mode = release
    proxy_pac = pac.lua
    network = public
    plat = windows
}
D:\projects\xmake\core\build\xmake.exe require -v -D -j 14 --extra={system=false} gflags
checking for git ... ok
checking for gzip ... ok
checking for tar ... ok
git rev-parse HEAD
checking for cl.exe ... C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\bin\HostX64\x64\cl.exe
checking for Microsoft Visual Studio (x64) version ... 2022
checking for Microsoft C/C++ Compiler (x64) version ... 19.42.34435
checkinfo: cannot runv(nim.exe --version), No such file or directory
checking for nim ... no
checkinfo: cannot runv(nim.exe --version), No such file or directory
checking for nim ... no
checking for cmake ... no
checking for cmake ... no
checking for cmake ... no
checking for cmake ... ok
finding gflags from xmake ..
checking for xmake::gflags ... no
note: install or modify (m) these packages (pass -y to skip confirm)?
in myrepo:
  -> gflags v2.2.2 [toolchains:"clang", runtimes:"MT", license:BSD-3-Clause]
please input: y (y/n/m)

checking for ping ... ok
pinging the host(github.com) ... 106 ms
tar -xf C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\v2.2.2.tar.gz
patching D:\projects\xmake-repo\packages\g\gflags\patches\v2.2.2\fix-cmake.patch to gflags-v2.2.2 ..
applying patch C:\Users\leemu\AppData\Local\Temp\.xmake\241216\patches\gflags\v2.2.2\fix-cmake.patch
git apply --reject --ignore-whitespace C:\Users\leemu\AppData\Local\Temp\.xmake\241216\patches\gflags\v2.2.2\fix-cmake.patch
Checking patch cmake/config.cmake.in...
Applied patch cmake/config.cmake.in cleanly.
checking for clang ... ok
checking for the c compiler (cc) ... clang
checking for clang ... ok
checking for the c++ compiler (cxx) ... clang
checkinfo: cannot runv(pkgconf.exe --version), No such file or directory
checking for pkgconf ... no
checkinfo: cannot runv(pkgconf.exe --version), No such file or directory
checking for pkgconf ... no
checking for pkg-config ... ok
checking for cmake ... ok
checking for flags (clang_ms_runtime_lib) ... ok
> clang "-fms-runtime-lib=dll" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
checking for flags (clang_ms_runtime_lib) ... ok
> clang "-fms-runtime-lib=dll" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
checking for the linker (ld) ... clang++
checking for clang++ ... ok
checking for flags (clang_ms_runtime_lib) ... ok
> clang++ "-fms-runtime-lib=dll" "-Qunused-arguments"
checking for ar ... ok
checking for the static library archiver (ar) ... ar
checking for the shared library linker (sh) ... clang++
checking for flags (clang_ms_runtime_lib) ... ok
> clang++ "-fms-runtime-lib=dll" "-Qunused-arguments"
checkinfo: cannot runv(pkgconf.exe --version), No such file or directory
checking for pkgconf ... no
checkinfo: cannot runv(pkgconf.exe --version), No such file or directory
checking for pkgconf ... no
checking for pkg-config ... ok
cmake -DBUILD_TESTING=OFF -DGFLAGS_REGISTER_BUILD_DIR=OFF -DGFLAGS_REGISTER_INSTALL_PREFIX=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DBUILD_STATIC_LIBS=ON -DBUILD_gflags_LIB=ON -DBUILD_gflags_nothreads_LIB=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_INSTALL_PREFIX=C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\0cd24927a324499db8337def71d37b3b -DCMAKE_INSTALL_LIBDIR:PATH=lib -G "Visual Studio 17 2022" -A x64 -DCMAKE_GENERATOR_TOOLSET=v143 -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded -DCMAKE_COMPILE_PDB_OUTPUT_DIRECTORY=C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_0cd24927/pdb -DCMAKE_PDB_OUTPUT_DIRECTORY=C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_0cd24927/pdb "-DCMAKE_CXX_FLAGS_RELEASE=/DWIN32 /D_WINDOWS /W3 /GR /EHsc  /O2 /Ob2 /DNDEBUG -fms-runtime-lib=static" "-DCMAKE_EXE_LINKER_FLAGS_RELEASE=/machine:x64 /INCREMENTAL:NO -nostdlib" -DCMAKE_STATIC_LINKER_FLAGS_RELEASE=/machine:x64 "-DCMAKE_MODULE_LINKER_FLAGS_RELEASE=  -nostdlib" "-DCMAKE_C_FLAGS_RELEASE=/DWIN32 /D_WINDOWS /W3  /O2 /Ob2 /DNDEBUG -fms-runtime-lib=static" "-DCMAKE_SHARED_LINKER_FLAGS_RELEASE=/machine:x64 /INCREMENTAL:NO -nostdlib" C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source
CMake Deprecation Warning at CMakeLists.txt:73 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Selecting Windows SDK version 10.0.26100.0 to target Windows 10.0.19045.
-- The CXX compiler identification is MSVC 19.42.34435.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.42.34433/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check size of __int32
-- Check size of __int32 - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
-- Found Threads: TRUE
-- Configuring done (4.1s)
-- Generating done (0.1s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_FLAGS_RELEASE
    CMAKE_INSTALL_LIBDIR


-- Build files have been written to: C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_0cd24927
checking for msbuild.exe ... C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Current\Bin\amd64\msbuild.exe
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Current\Bin\amd64\msbuild.exe ALL_BUILD.vcxproj -t:Rebuild /nr:false -nologo -m:14 -p:Configuration=Release -p:Platform=x64 /p:PlatformToolset=v143
生成启动时间为 2024/12/16 15:25:51。

     1>项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ALL_BUILD.vcxproj” 在节
       点 1 上(Rebuild 个目标)。
     1>CoreClean:
         正在创建目录“x64\Release\ALL_BUILD\”。
     1>项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ALL_BUILD.vcxproj”(1
       )正在节点 1 上生成“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ZERO_CHECK.
       vcxproj”(2) (Clean 个目标)。
     2>CoreClean:
         正在创建目录“x64\Release\ZERO_CHECK\”。
     2>已完成生成项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ZERO_CHECK.vcxp
       roj”(Clean 个目标)的操作。
     1>项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ALL_BUILD.vcxproj”(1
       )正在节点 2 上生成“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\gflags_stat
       ic.vcxproj”(3) (Clean 个目标)。
     3>CoreClean:
         正在创建目录“gflags_static.dir\Release\”。
     3>已完成生成项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\gflags_static.v
       cxproj”(Clean 个目标)的操作。
     1>项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ALL_BUILD.vcxproj”(1
       )正在节点 1 上生成“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ZERO_CHECK.
       vcxproj”(2:3) (默认目标)。
     2>PrepareForBuild:
         Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://
         aka.ms/cpp/structured-output for more details.
         正在创建目录“x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\”。
       InitializeBuildStatus:
         正在创建“x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild”，因为已指定“AlwaysCreate”。
         正在对“x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild”执行 Touch 任务。
       CustomBuild:
         1>Checking Build System
       FinalizeBuildStatus:
         正在删除文件“x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild”。
         正在对“x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate”执行 Touch 任务。
     2>已完成生成项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ZERO_CHECK.vcxp
       roj”(默认目标)的操作。
     1>项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ALL_BUILD.vcxproj”(1
       )正在节点 2 上生成“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\gflags_stat
       ic.vcxproj”(3:2) (默认目标)。
     3>PrepareForBuild:
         Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://
         aka.ms/cpp/structured-output for more details.
         正在创建目录“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\lib\Release\”。
         正在创建目录“gflags_static.dir\Release\gflags_static.tlog\”。
       InitializeBuildStatus:
         正在创建“gflags_static.dir\Release\gflags_static.tlog\unsuccessfulbuild”，因为已指定“AlwaysCreate”。
         正在对“gflags_static.dir\Release\gflags_static.tlog\unsuccessfulbuild”执行 Touch 任务。
       CustomBuild:
         Building Custom Rule C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/CMakeLists.txt
       MakeDirsForCl:
         正在创建目录“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\pdb\Release”。
       ClCompile:
         C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\bin\HostX64\x64\CL.exe /c
         /IC:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\include /IC:\Users\
         leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\src /IC:\Users\leemu\AppData\Local\.xmake\cac
         he\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\include\gflags /nologo /W3 /WX- /diagnostics:column /O2 /Ob2
          /D _MBCS /D WIN32 /D _WINDOWS /D NDEBUG /D GFLAGS_IS_A_DLL=0 /D "CMAKE_INTDIR=\"Release\"" /EHsc /GS /fp:precise /
         Zc:wchar_t /Zc:forScope /Zc:inline /GR /Fo"gflags_static.dir\Release\\" /Fd"C:\Users\leemu\AppData\Local\.xmake\cac
         he\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\pdb\Release\\" /external:W3 /Gd /TP /errorReport:queue  -fms
         -runtime-lib=static C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\src\gflags.cc C:
         \Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\src\gflags_reporting.cc C:\Users\leemu
         \AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\src\gflags_completions.cc C:\Users\leemu\AppData\L
         ocal\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\src\windows_port.cc
     3>cl : command line  warning D9002: ignoring unknown option '-fms-runtime-lib=static' [C:\Users\leemu\AppData\Local\.xm
       ake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\gflags_static.vcxproj]
         gflags.cc
         gflags_reporting.cc
         gflags_completions.cc
         windows_port.cc
         Generating Code...
       Lib:
         C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\bin\HostX64\x64\Lib.exe /O
         UT:"C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\lib\Release\gflag
         s_static.lib" /NOLOGO /MACHINE:X64  /machine:x64 /machine:x64 gflags_static.dir\Release\gflags.obj
         gflags_static.dir\Release\gflags_reporting.obj
         gflags_static.dir\Release\gflags_completions.obj
         gflags_static.dir\Release\windows_port.obj
         gflags_static.vcxproj -> C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24
         927\lib\Release\gflags_static.lib
       FinalizeBuildStatus:
         正在删除文件“gflags_static.dir\Release\gflags_static.tlog\unsuccessfulbuild”。
         正在对“gflags_static.dir\Release\gflags_static.tlog\gflags_static.lastbuildstate”执行 Touch 任务。
     3>已完成生成项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\gflags_static.v
       cxproj”(默认目标)的操作。
     1>PrepareForBuild:
         Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://
         aka.ms/cpp/structured-output for more details.
         正在创建目录“x64\Release\ALL_BUILD\ALL_BUILD.tlog\”。
       InitializeBuildStatus:
         正在创建“x64\Release\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild”，因为已指定“AlwaysCreate”。
         正在对“x64\Release\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild”执行 Touch 任务。
       CustomBuild:
         Building Custom Rule C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/CMakeLists.txt
       FinalizeBuildStatus:
         正在删除文件“x64\Release\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild”。
         正在对“x64\Release\ALL_BUILD\ALL_BUILD.tlog\ALL_BUILD.lastbuildstate”执行 Touch 任务。
     1>已完成生成项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ALL_BUILD.vcxpr
       oj”(Rebuild 个目标)的操作。

已成功生成。

       “C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ALL_BUILD.vcxproj”(Reb
       uild 目标) (1) ->
       “C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\gflags_static.vcxproj”
       (默认目标) (3:2) ->
       (ClCompile 目标) ->
         cl : command line  warning D9002: ignoring unknown option '-fms-runtime-lib=static' [C:\Users\leemu\AppData\Local\.
       xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\gflags_static.vcxproj]

    1 个警告
    0 个错误

已用时间 00:00:03.87
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Current\Bin\amd64\msbuild.exe INSTALL.vcxproj -nologo -m:14 -p:Configuration=Release -p:Platform=x64 /p:PlatformToolset=v143
生成启动时间为 2024/12/16 15:25:55。

     1>节点 1 上的项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\INSTALL.vcxpr
       oj”(默认目标)。
     1>项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\INSTALL.vcxproj”(1)正
       在节点 1 上生成“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ZERO_CHECK.vc
       xproj”(2) (默认目标)。
     2>PrepareForBuild:
         Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://
         aka.ms/cpp/structured-output for more details.
       InitializeBuildStatus:
         正在创建“x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild”，因为已指定“AlwaysCreate”。
         正在对“x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild”执行 Touch 任务。
       CustomBuild:
         所有输出均为最新。
         1>Checking Build System
       FinalizeBuildStatus:
         正在删除文件“x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild”。
         正在对“x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate”执行 Touch 任务。
     2>已完成生成项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ZERO_CHECK.vcxp
       roj”(默认目标)的操作。
     1>项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\INSTALL.vcxproj”(1)正
       在节点 2 上生成“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ALL_BUILD.vcx
       proj”(3) (默认目标)。
     3>项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ALL_BUILD.vcxproj”(3
       )正在节点 2 上生成“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\gflags_stat
       ic.vcxproj”(4) (默认目标)。
     4>PrepareForBuild:
         Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://
         aka.ms/cpp/structured-output for more details.
       InitializeBuildStatus:
         正在创建“gflags_static.dir\Release\gflags_static.tlog\unsuccessfulbuild”，因为已指定“AlwaysCreate”。
         正在对“gflags_static.dir\Release\gflags_static.tlog\unsuccessfulbuild”执行 Touch 任务。
       CustomBuild:
         所有输出均为最新。
         Building Custom Rule C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/CMakeLists.txt
       ClCompile:
         所有输出均为最新。
       Lib:
         所有输出均为最新。
         gflags_static.vcxproj -> C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24
         927\lib\Release\gflags_static.lib
       FinalizeBuildStatus:
         正在删除文件“gflags_static.dir\Release\gflags_static.tlog\unsuccessfulbuild”。
         正在对“gflags_static.dir\Release\gflags_static.tlog\gflags_static.lastbuildstate”执行 Touch 任务。
     4>已完成生成项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\gflags_static.v
       cxproj”(默认目标)的操作。
     3>PrepareForBuild:
         Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://
         aka.ms/cpp/structured-output for more details.
       InitializeBuildStatus:
         正在创建“x64\Release\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild”，因为已指定“AlwaysCreate”。
         正在对“x64\Release\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild”执行 Touch 任务。
       CustomBuild:
         所有输出均为最新。
         Building Custom Rule C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/CMakeLists.txt
       FinalizeBuildStatus:
         正在删除文件“x64\Release\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild”。
         正在对“x64\Release\ALL_BUILD\ALL_BUILD.tlog\ALL_BUILD.lastbuildstate”执行 Touch 任务。
     3>已完成生成项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\ALL_BUILD.vcxpr
       oj”(默认目标)的操作。
     1>PrepareForBuild:
         正在创建目录“x64\Release\INSTALL\”。
         Structured output is enabled. The formatting of compiler diagnostics will reflect the error hierarchy. See https://
         aka.ms/cpp/structured-output for more details.
         正在创建目录“x64\Release\INSTALL\INSTALL.tlog\”。
       InitializeBuildStatus:
         正在创建“x64\Release\INSTALL\INSTALL.tlog\unsuccessfulbuild”，因为已指定“AlwaysCreate”。
         正在对“x64\Release\INSTALL\INSTALL.tlog\unsuccessfulbuild”执行 Touch 任务。
       CustomBuild:
         1>
       PostBuildEvent:
         setlocal
         D:\Applications\Scoop\apps\cmake\3.31.1\bin\cmake.exe -DBUILD_TYPE=Release -P cmake_install.cmake
         if %errorlevel% neq 0 goto :cmEnd
         :cmEnd
         endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone
         :cmErrorLevel
         exit /b %1
         :cmDone
         if %errorlevel% neq 0 goto :VCEnd
         :VCEnd
         -- Install configuration: "Release"
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/lib/gf
         lags_static.lib
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/includ
         e/gflags/gflags.h
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/includ
         e/gflags/gflags_declare.h
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/includ
         e/gflags/gflags_completions.h
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/includ
         e/gflags/gflags_gflags.h
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/lib/cm
         ake/gflags/gflags-config.cmake
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/lib/cm
         ake/gflags/gflags-config-version.cmake
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/lib/cm
         ake/gflags/gflags-targets.cmake
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/lib/cm
         ake/gflags/gflags-targets-release.cmake
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/lib/cm
         ake/gflags/gflags-nonamespace-targets.cmake
         -- Installing: C:/Users/leemu/AppData/Local/.xmake/packages/g/gflags/v2.2.2/0cd24927a324499db8337def71d37b3b/lib/cm
         ake/gflags/gflags-nonamespace-targets-release.cmake
       FinalizeBuildStatus:
         正在删除文件“x64\Release\INSTALL\INSTALL.tlog\unsuccessfulbuild”。
         正在对“x64\Release\INSTALL\INSTALL.tlog\INSTALL.lastbuildstate”执行 Touch 任务。
     1>已完成生成项目“C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_0cd24927\INSTALL.vcxproj
       ”(默认目标)的操作。

已成功生成。
    0 个警告
    0 个错误

已用时间 00:00:01.03
finding gflags from xmake ..
checking for xmake::gflags ... gflags v2.2.2
{
  license = "BSD-3-Clause",
  links = {
    "gflags_static"
  },
  static = true,
  syslinks = "shlwapi",
  sysincludedirs = {
    "C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\0cd24927a324499db8337def71d37b3b\include"
  },
  libfiles = {
    "C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\0cd24927a324499db8337def71d37b3b\lib\gflags_static.lib"
  },
  linkdirs = {
    "C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\0cd24927a324499db8337def71d37b3b\lib"
  },
  version = "v2.2.2"
}

patching C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\0cd24927a324499db8337def71d37b3b\lib\pkgconfig\gflags.pc ..
checking for flags (-fms-runtime-lib=static) ... ok
> clang "-fms-runtime-lib=static" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
> clang -c -Qunused-arguments -m64 --target=x86_64-windows-msvc -fms-runtime-lib=static -std=c++11 -isystem C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\0cd24927a324499db8337def71d37b3b\include -o C:\Users\leemu\AppData\Local\Temp\.xmake\241216\_04D7906943E042108A33EF73CF8A9930.o C:\Users\leemu\AppData\Local\Temp\.xmake\241216\_BAAC64F40B604011AD0907FF91B9003A.cpp
checking for flags (-fdiagnostics-color=always) ... ok
> clang "-fdiagnostics-color=always" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
checking for flags (-fansi-escape-codes) ... ok
> clang "-fansi-escape-codes" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
checking for flags (-Wno-gnu-line-marker -Werror) ... ok
> clang "-Wno-gnu-line-marker" "-Werror" "-Qunused-arguments" "-m64" "--target=x86_64-windows-msvc"
> checking for c++ includes(gflags/gflags.h)
> checking for c++ links(gflags_static, shlwapi)
> checking for c++ snippet(test)
checkinfo: ...amdir\core\sandbox\modules\import\core\tool\compiler.lua:84: @programdir\modules\core\tools\gcc.lua:966: In file included from C:\Users\leemu\AppData\Local\Temp\.xmake\241216\_BAAC64F40B604011AD0907FF91B9003A.cpp:2:
C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\0cd24927a324499db8337def71d37b3b\include/gflags/gflags.h:81:10: fatal error: 'string' file not found
   81 | #include <string>
      |          ^~~~~~~~
1 error generated.
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\modules\core\tools\gcc.lua:966]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [@programdir\modules\core\tools\gcc.lua:908]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]:
    [@programdir\core\tool\compiler.lua:288]: in function 'compile'
    [...amdir\core\sandbox\modules\import\core\tool\compiler.lua:82]: in function 'compile'
    [@programdir\modules\lib\detect\check_cxsnippets.lua:244]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:117]: in function 'try'
    [@programdir\modules\lib\detect\check_cxsnippets.lua:237]:
    [D:\projects\xmake-repo\packages\g\gflags\xmake.lua:48]: in function 'script'
    [...dir\modules\private\action\require\impl\utils\filter.lua:114]: in function 'call'
    [...dir\modules\private\action\require\impl\actions\test.lua:41]:
    [...\modules\private\action\require\impl\actions\install.lua:495]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:117]: in function 'try'
    [...\modules\private\action\require\impl\actions\install.lua:419]:
    [...modules\private\action\require\impl\install_packages.lua:510]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:117]: in function 'try'
    [@programdir\modules\async\runjobs.lua:224]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:406]:

error: D:\projects\xmake-repo\packages\g\gflags\xmake.lua:48: ...amdir\core\sandbox\modules\import\core\tool\compiler.lua:84: @programdir\modules\core\tools\gcc.lua:966: In file included from C:\Users\leemu\AppData\Local\Temp\.xmake\241216\_BAAC64F40B604011AD0907FF91B9003A.cpp:2:
C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\0cd24927a324499db8337def71d37b3b\include/gflags/gflags.h:81:10: fatal error: 'string' file not found
   81 | #include <string>
      |          ^~~~~~~~
1 error generated.
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\modules\core\tools\gcc.lua:966]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [@programdir\modules\core\tools\gcc.lua:908]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]:
    [@programdir\core\tool\compiler.lua:288]: in function 'compile'
    [...amdir\core\sandbox\modules\import\core\tool\compiler.lua:82]: in function 'compile'
    [@programdir\modules\lib\detect\check_cxsnippets.lua:244]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:117]: in function 'try'
    [@programdir\modules\lib\detect\check_cxsnippets.lua:237]:
    [D:\projects\xmake-repo\packages\g\gflags\xmake.lua:48]: in function 'script'
    [...dir\modules\private\action\require\impl\utils\filter.lua:114]: in function 'call'
    [...dir\modules\private\action\require\impl\actions\test.lua:41]:
    [...\modules\private\action\require\impl\actions\install.lua:495]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:117]: in function 'try'
    [...\modules\private\action\require\impl\actions\install.lua:419]:
    [...modules\private\action\require\impl\install_packages.lua:510]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:244]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:117]: in function 'try'
    [@programdir\modules\async\runjobs.lua:224]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:406]:

  => install gflags v2.2.2 .. failed
error: @programdir\core\main.lua:329: @programdir\modules\async\runjobs.lua:325: ...\modules\private\action\require\impl\actions\install.lua:561: install failed!
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [...\modules\private\action\require\impl\actions\install.lua:561]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [...\modules\private\action\require\impl\actions\install.lua:419]:
    [...modules\private\action\require\impl\install_packages.lua:510]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:1004: in function 'base/os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:329: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
error: .\xmake\core\main.lua:329: .\xmake\core\sandbox\modules\os.lua:378: execv(D:\projects\xmake\core\build\xmake.exe require -v -D -j 14 --extra={system=false} gflags) failed(-1)
stack traceback:
    [C]: in function 'error'
    [.\xmake\core\base\os.lua:1004]:
    [.\xmake\core\sandbox\modules\os.lua:378]:
    [.\xmake\modules\private\xrepo\action\install.lua:309]: in function '_install_packages'
    [.\xmake\modules\private\xrepo\action\install.lua:316]:
    [.\xmake\modules\private\xrepo\main.lua:196]:
    [.\xmake\plugins\lua\main.lua:124]:
    [C]: in function 'xpcall'
    [.\xmake\core\base\utils.lua:244]:
    [.\xmake\core\base\task.lua:491]: in function 'run'
    [.\xmake\core\main.lua:327]: in function 'cotask'
    [.\xmake\core\base\scheduler.lua:406]:

stack traceback:
        [C]: in function 'error'
        .\xmake\core\base\os.lua:1004: in function 'os.raiselevel'
        (...tail calls...)
        .\xmake\core\main.lua:329: in upvalue 'cotask'
        .\xmake\core\base\scheduler.lua:406: in function <.\xmake\core\base\scheduler.lua:399>

```
</details>

不知道为什么，单纯 `-p mingw` ，安装基于 CMake 的包，会传递 `ar` 的绝对路径，即 `-DCMAKE_AR=/path/to/ar` 而 `-p mingw --toolchain=clang` 则传递 `-DCMAKE_AR=ar.exe` ，然后 CMake 就找不到 `ar` 了

<details>
<summary> xrepo install -vD -p mingw --toolchain=clang gflags （安装失败）</summary>

```
D:\projects\xmake\core\build\xmake.exe f -c --require=n -vD -p mingw --toolchain=clang
checking for architecture ... x86_64
checking for clang ... ok
configure
{
    theme = default
    clean = true
    mingw = D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/
    network = public
    plat = mingw
    buildir = build
    ndk_stdcxx = true
    ccache = true
    proxy = socks5://127.0.0.1:20170
    proxy_pac = pac.lua
    kind = static
    toolchain = clang
    mode = release
    arch = x86_64
    host = windows
}
D:\projects\xmake\core\build\xmake.exe require -v -D -j 14 --extra={system=false} gflags
checking for git ... ok
checking for gzip ... ok
checking for tar ... ok
git rev-parse HEAD
checking for cl.exe ... C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\bin\HostX64\x64\cl.exe
checking for Microsoft Visual Studio (x64) version ... 2022
checking for Microsoft C/C++ Compiler (x64) version ... 19.42.34435
checkinfo: cannot runv(nim.exe --version), No such file or directory
checking for nim ... no
checkinfo: cannot runv(nim.exe --version), No such file or directory
checking for nim ... no
checking for cmake ... no
checking for cmake ... no
checking for cmake ... no
checking for cmake ... ok
finding gflags from xmake ..
checking for xmake::gflags ... no
note: install or modify (m) these packages (pass -y to skip confirm)?
in myrepo:
  -> gflags v2.2.2 [toolchains:"clang", license:BSD-3-Clause]
please input: y (y/n/m)

checking for ping ... ok
pinging the host(github.com) ... 105 ms
tar -xf C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\v2.2.2.tar.gz
patching D:\projects\xmake-repo\packages\g\gflags\patches\v2.2.2\fix-cmake.patch to gflags-v2.2.2 ..
applying patch C:\Users\leemu\AppData\Local\Temp\.xmake\241216\patches\gflags\v2.2.2\fix-cmake.patch
git apply --reject --ignore-whitespace C:\Users\leemu\AppData\Local\Temp\.xmake\241216\patches\gflags\v2.2.2\fix-cmake.patch
Checking patch cmake/config.cmake.in...
Applied patch cmake/config.cmake.in cleanly.
checking for clang ... ok
checking for the c compiler (cc) ... clang
checking for clang ... ok
checking for the c++ compiler (cxx) ... clang
checking for ninja ... ok
checking for clang ... ok
checking for the assember (as) ... clang
checking for ar ... ok
checking for the static library archiver (ar) ... ar
checking for ranlib ... ok
checking for the archive index generator (ranlib) ... ranlib
checking for llvm-rc ... ok
checking for the windows resource compiler (mrc) ... llvm-rc
checking for clang++ ... ok
checking for the linker (ld) ... clang++
checking for clang++ ... ok
checking for the shared library linker (sh) ... clang++
checking for pkg-config ... ok
checking for cmake ... ok
checking for pkg-config ... ok
cmake -DBUILD_TESTING=OFF -DGFLAGS_REGISTER_BUILD_DIR=OFF -DGFLAGS_REGISTER_INSTALL_PREFIX=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DBUILD_STATIC_LIBS=ON -DBUILD_gflags_LIB=ON -DBUILD_gflags_nothreads_LIB=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_INSTALL_PREFIX=C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\6758891cfbde4f4cb92634e6368a5498 -DCMAKE_INSTALL_LIBDIR:PATH=lib -G Ninja -DCMAKE_FIND_ROOT_PATH=D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/ "-DCMAKE_EXE_LINKER_FLAGS=-m64 --target=x86_64-w64-windows-gnu" "-DCMAKE_ASM_FLAGS=-m64 --target=x86_64-w64-windows-gnu" -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=BOTH "-DCMAKE_C_FLAGS=-m64 --target=x86_64-w64-windows-gnu" -DCMAKE_CXX_COMPILER=clang++.exe "-DCMAKE_CXX_FLAGS=-m64 --target=x86_64-w64-windows-gnu" -DHAVE_FLAG_SEARCH_PATHS_FIRST=0 -DCMAKE_OSX_SYSROOT= -DCMAKE_SYSTEM_PROCESSOR=x86_64 -DCMAKE_C_COMPILER=clang.exe -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=BOTH -DCMAKE_ASM_COMPILER=clang.exe "-DCMAKE_MODULE_LINKER_FLAGS=-m64 --target=x86_64-w64-windows-gnu" "-DCMAKE_SHARED_LINKER_FLAGS=-m64 --target=x86_64-w64-windows-gnu" -DCMAKE_MAKE_PROGRAM=ninja -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=BOTH -DCMAKE_AR=ar.exe -DCMAKE_RANLIB=ranlib.exe -DCMAKE_RC_COMPILER=llvm-rc.exe -DCMAKE_STATIC_LINKER_FLAGS= "-DCMAKE_CXX_FLAGS_RELEASE=-m64 --target=x86_64-w64-windows-gnu -O3 -DNDEBUG" -DCMAKE_STATIC_LINKER_FLAGS_RELEASE= "-DCMAKE_EXE_LINKER_FLAGS_RELEASE=-m64 --target=x86_64-w64-windows-gnu" "-DCMAKE_C_FLAGS_RELEASE=-m64 --target=x86_64-w64-windows-gnu -O3 -DNDEBUG" "-DCMAKE_SHARED_LINKER_FLAGS_RELEASE=-m64 --target=x86_64-w64-windows-gnu" C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source
CMake Deprecation Warning at CMakeLists.txt:73 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The CXX compiler identification is Clang 19.1.5
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/bin/clang++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include unistd.h
-- Looking for C++ include unistd.h - found
-- Looking for C++ include stdint.h
-- Looking for C++ include stdint.h - found
-- Looking for C++ include inttypes.h
-- Looking for C++ include inttypes.h - found
-- Looking for C++ include sys/types.h
-- Looking for C++ include sys/types.h - found
-- Looking for C++ include sys/stat.h
-- Looking for C++ include sys/stat.h - found
-- Looking for C++ include fnmatch.h
-- Looking for C++ include fnmatch.h - not found
-- Looking for C++ include shlwapi.h
-- Looking for C++ include shlwapi.h - found
-- Looking for C++ include stddef.h
-- Looking for C++ include stddef.h - found
-- Check size of uint32_t
-- Check size of uint32_t - done
-- Looking for strtoll
-- Looking for strtoll - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Check size of pthread_rwlock_t
-- Check size of pthread_rwlock_t - failed
-- Configuring done (6.1s)
-- Generating done (0.1s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_ASM_COMPILER
    CMAKE_ASM_FLAGS
    CMAKE_C_COMPILER
    CMAKE_C_FLAGS
    CMAKE_C_FLAGS_RELEASE
    CMAKE_FIND_ROOT_PATH
    CMAKE_FIND_ROOT_PATH_MODE_LIBRARY
    CMAKE_INSTALL_LIBDIR
    HAVE_FLAG_SEARCH_PATHS_FIRST


-- Build files have been written to: C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c
checking for pkg-config ... ok
ninja install -C C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_6758891c -v -j 14
ninja: Entering directory `C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_6758891c'
[1/6] D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\clang++.exe -DGFLAGS_IS_A_DLL=0 -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c/include -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/src -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c/include/gflags -m64 --target=x86_64-w64-windows-gnu -m64 --target=x86_64-w64-windows-gnu -O3 -DNDEBUG -MD -MT CMakeFiles/gflags_static.dir/src/windows_port.cc.obj -MF CMakeFiles\gflags_static.dir\src\windows_port.cc.obj.d -o CMakeFiles/gflags_static.dir/src/windows_port.cc.obj -c C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/src/windows_port.cc
[2/6] D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\clang++.exe -DGFLAGS_IS_A_DLL=0 -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c/include -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/src -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c/include/gflags -m64 --target=x86_64-w64-windows-gnu -m64 --target=x86_64-w64-windows-gnu -O3 -DNDEBUG -MD -MT CMakeFiles/gflags_static.dir/src/gflags_reporting.cc.obj -MF CMakeFiles\gflags_static.dir\src\gflags_reporting.cc.obj.d -o CMakeFiles/gflags_static.dir/src/gflags_reporting.cc.obj -c C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/src/gflags_reporting.cc
[3/6] D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\clang++.exe -DGFLAGS_IS_A_DLL=0 -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c/include -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/src -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c/include/gflags -m64 --target=x86_64-w64-windows-gnu -m64 --target=x86_64-w64-windows-gnu -O3 -DNDEBUG -MD -MT CMakeFiles/gflags_static.dir/src/gflags_completions.cc.obj -MF CMakeFiles\gflags_static.dir\src\gflags_completions.cc.obj.d -o CMakeFiles/gflags_static.dir/src/gflags_completions.cc.obj -c C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/src/gflags_completions.cc
[4/6] D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\clang++.exe -DGFLAGS_IS_A_DLL=0 -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c/include -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/src -IC:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/build_6758891c/include/gflags -m64 --target=x86_64-w64-windows-gnu -m64 --target=x86_64-w64-windows-gnu -O3 -DNDEBUG -MD -MT CMakeFiles/gflags_static.dir/src/gflags.cc.obj -MF CMakeFiles\gflags_static.dir\src\gflags.cc.obj.d -o CMakeFiles/gflags_static.dir/src/gflags.cc.obj -c C:/Users/leemu/AppData/Local/.xmake/cache/packages/2412/g/gflags/v2.2.2/source/src/gflags.cc
[5/6] C:\Windows\system32\cmd.exe /C "cd . && D:\Applications\Scoop\apps\cmake\3.31.1\bin\cmake.exe -E rm -f lib\libgflags_static.a && C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_6758891c\ar.exe qc lib\libgflags_static.a  CMakeFiles/gflags_static.dir/src/gflags.cc.obj CMakeFiles/gflags_static.dir/src/gflags_reporting.cc.obj CMakeFiles/gflags_static.dir/src/gflags_completions.cc.obj CMakeFiles/gflags_static.dir/src/windows_port.cc.obj && C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_6758891c\ranlib.exe lib\libgflags_static.a && cd ."
FAILED: lib/libgflags_static.a
C:\Windows\system32\cmd.exe /C "cd . && D:\Applications\Scoop\apps\cmake\3.31.1\bin\cmake.exe -E rm -f lib\libgflags_static.a && C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_6758891c\ar.exe qc lib\libgflags_static.a  CMakeFiles/gflags_static.dir/src/gflags.cc.obj CMakeFiles/gflags_static.dir/src/gflags_reporting.cc.obj CMakeFiles/gflags_static.dir/src/gflags_completions.cc.obj CMakeFiles/gflags_static.dir/src/windows_port.cc.obj && C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_6758891c\ranlib.exe lib\libgflags_static.a && cd ."
'C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_6758891c\ar.exe' is not recognized as an internal or external command,
operable program or batch file.
ninja: build stopped: subcommand failed.
error: @programdir\core\sandbox\modules\os.lua:378: execv(ninja install -C C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source\build_6758891c -v -j 14) failed(1)
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\core\sandbox\modules\os.lua:378]:
    [@programdir\core\sandbox\modules\os.lua:291]: in function 'vrunv'
    [@programdir\modules\package\tools\ninja.lua:64]: in function 'install'
    [@programdir\modules\package\tools\cmake.lua:1168]: in function '_install_for_ninja'
    [@programdir\modules\package\tools\cmake.lua:1369]: in function 'install'
    [D:\projects\xmake-repo\packages\g\gflags\xmake.lua:44]: in function 'script'
    [...dir\modules\private\action\require\impl\utils\filter.lua:114]: in function 'call'
    [...\modules\private\action\require\impl\actions\install.lua:452]:

  => install gflags v2.2.2 .. failed
error: @programdir\core\main.lua:329: @programdir\modules\async\runjobs.lua:325: ...\modules\private\action\require\impl\actions\install.lua:561: install failed!
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [...\modules\private\action\require\impl\actions\install.lua:561]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [...\modules\private\action\require\impl\actions\install.lua:419]:
    [...modules\private\action\require\impl\install_packages.lua:510]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:1004: in function 'os.raiselevel'
        (...tail calls...)
        @programdir\core\main.lua:329: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>
error: .\xmake\core\main.lua:329: .\xmake\core\sandbox\modules\os.lua:378: execv(D:\projects\xmake\core\build\xmake.exe require -v -D -j 14 --extra={system=false} gflags) failed(-1)
stack traceback:
    [C]: in function 'error'
    [.\xmake\core\base\os.lua:1004]:
    [.\xmake\core\sandbox\modules\os.lua:378]:
    [.\xmake\modules\private\xrepo\action\install.lua:309]: in function '_install_packages'
    [.\xmake\modules\private\xrepo\action\install.lua:316]:
    [.\xmake\modules\private\xrepo\main.lua:196]:
    [.\xmake\plugins\lua\main.lua:124]:
    [C]: in function 'xpcall'
    [.\xmake\core\base\utils.lua:244]:
    [.\xmake\core\base\task.lua:491]: in function 'run'
    [.\xmake\core\main.lua:327]: in function 'cotask'
    [.\xmake\core\base\scheduler.lua:406]:

stack traceback:
        [C]: in function 'error'
        .\xmake\core\base\os.lua:1004: in function 'os.raiselevel'
        (...tail calls...)
        .\xmake\core\main.lua:329: in upvalue 'cotask'
        .\xmake\core\base\scheduler.lua:406: in function <.\xmake\core\base\scheduler.lua:399>
```

</details>

<details>
<summary> xrepo install -vD -p mingw gflags</summary>

```
D:\projects\xmake\core\build\xmake.exe f -c --require=n -vD -p mingw
checking for architecture ... x86_64
checking for mingw directory ... D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/
configure
{
    arch = x86_64
    proxy = socks5://127.0.0.1:20170
    kind = static
    proxy_pac = pac.lua
    clean = true
    host = windows
    ndk_stdcxx = true
    plat = mingw
    theme = default
    network = public
    mingw = D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/
    buildir = build
    ccache = true
    mode = release
}
D:\projects\xmake\core\build\xmake.exe require -v -D -j 14 --extra={system=false} gflags
checking for git ... ok
checking for gzip ... ok
checking for tar ... ok
git rev-parse HEAD
checking for cl.exe ... C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.42.34433\bin\HostX64\x64\cl.exe
checking for Microsoft Visual Studio (x64) version ... 2022
checking for Microsoft C/C++ Compiler (x64) version ... 19.42.34435
checkinfo: cannot runv(nim.exe --version), No such file or directory
checking for nim ... no
checkinfo: cannot runv(nim.exe --version), No such file or directory
checking for nim ... no
checking for cmake ... no
checking for cmake ... no
checking for cmake ... no
checking for cmake ... ok
finding gflags from xmake ..
checking for xmake::gflags ... no
note: install or modify (m) these packages (pass -y to skip confirm)?
in myrepo:
  -> gflags v2.2.2 [license:BSD-3-Clause]
please input: y (y/n/m)

checking for ping ... ok
pinging the host(github.com) ... 105 ms
tar -xf C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\v2.2.2.tar.gz
patching D:\projects\xmake-repo\packages\g\gflags\patches\v2.2.2\fix-cmake.patch to gflags-v2.2.2 ..
applying patch C:\Users\leemu\AppData\Local\Temp\.xmake\241216\patches\gflags\v2.2.2\fix-cmake.patch
git apply --reject --ignore-whitespace C:\Users\leemu\AppData\Local\Temp\.xmake\241216\patches\gflags\v2.2.2\fix-cmake.patch
Checking patch cmake/config.cmake.in...
Applied patch cmake/config.cmake.in cleanly.
checking for x86_64-w64-mingw32-gcc ... D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\x86_64-w64-mingw32-gcc
checking for the c compiler (cc) ... x86_64-w64-mingw32-gcc
checking for x86_64-w64-mingw32-g++ ... D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\x86_64-w64-mingw32-g++
checking for the c++ compiler (cxx) ... x86_64-w64-mingw32-g++
checking for ninja ... ok
checking for x86_64-w64-mingw32-gcc ... D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\x86_64-w64-mingw32-gcc
checking for the assember (as) ... x86_64-w64-mingw32-gcc
checking for D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\ar ... ok
checking for the static library archiver (ar) ... ar
checking for D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\ranlib ... ok
checking for the archive index generator (ranlib) ... ranlib
checking for x86_64-w64-mingw32-windres ... D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\x86_64-w64-mingw32-windres
checking for the windows resource compiler (mrc) ... x86_64-w64-mingw32-windres
checking for x86_64-w64-mingw32-g++ ... D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\x86_64-w64-mingw32-g++
checking for the linker (ld) ... x86_64-w64-mingw32-g++
checking for x86_64-w64-mingw32-g++ ... D:\Applications\Scoop\apps\mingw-mstorsjo-llvm-ucrt\current\bin\x86_64-w64-mingw32-g++
checking for the shared library linker (sh) ... x86_64-w64-mingw32-g++
checking for pkg-config ... ok
checking for cmake ... ok
checking for pkg-config ... ok
cmake -DBUILD_TESTING=OFF -DGFLAGS_REGISTER_BUILD_DIR=OFF -DGFLAGS_REGISTER_INSTALL_PREFIX=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DBUILD_STATIC_LIBS=ON -DBUILD_gflags_LIB=ON -DBUILD_gflags_nothreads_LIB=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_INSTALL_PREFIX=C:\Users\leemu\AppData\Local\.xmake\packages\g\gflags\v2.2.2\af22f3785e204d19aaf8d28f0968a301 -DCMAKE_INSTALL_LIBDIR:PATH=lib -G Ninja -DCMAKE_SHARED_LINKER_FLAGS=-m64 -DCMAKE_CXX_FLAGS=-m64 -DCMAKE_ASM_FLAGS=-m64 -DCMAKE_RC_COMPILER=D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/bin/x86_64-w64-mingw32-windres.exe -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER -DCMAKE_CXX_COMPILER=D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/bin/x86_64-w64-mingw32-g++.exe -DCMAKE_EXE_LINKER_FLAGS=-m64 -DCMAKE_SYSTEM_PROCESSOR=x86_64 -DCMAKE_RANLIB=D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/bin/ranlib.exe -DCMAKE_MAKE_PROGRAM=ninja -DCMAKE_STATIC_LINKER_FLAGS= -DHAVE_FLAG_SEARCH_PATHS_FIRST=0 -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=BOTH -DCMAKE_MODULE_LINKER_FLAGS=-m64 -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=BOTH -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=BOTH -DCMAKE_FIND_ROOT_PATH=D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/ -DCMAKE_ASM_COMPILER=D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/bin/x86_64-w64-mingw32-gcc.exe -DCMAKE_C_FLAGS=-m64 -DCMAKE_AR=D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/bin/ar.exe -DCMAKE_C_COMPILER=D:/Applications/Scoop/apps/mingw-mstorsjo-llvm-ucrt/current/bin/x86_64-w64-mingw32-gcc.exe -DCMAKE_EXE_LINKER_FLAGS_RELEASE=-m64 -DCMAKE_STATIC_LINKER_FLAGS_RELEASE= "-DCMAKE_CXX_FLAGS_RELEASE=-m64 -O3 -DNDEBUG" -DCMAKE_SHARED_LINKER_FLAGS_RELEASE=-m64 "-DCMAKE_C_FLAGS_RELEASE=-m64 -O3 -DNDEBUG" C:\Users\leemu\AppData\Local\.xmake\cache\packages\2412\g\gflags\v2.2.2\source
```
（正常安装，以下省略）
</details>

--- a/xmake/toolchains/clang/load.lua
+++ b/xmake/toolchains/clang/load.lua
@@ -55,13 +55,13 @@ function main(toolchain, suffix)
     if toolchain:is_arch("x86_64", "x64") then
         target = "x86_64"
         march = "-m64"
-    elseif toolchain:is_arch("i386", "x86") then
-        target = "i386"
+    elseif toolchain:is_arch("i386", "x86", "i686") then
+        target = "i686"
         march = "-m32"
-    elseif toolchain:is_arch("arm64") then
-        target = "arm64"
+    elseif toolchain:is_arch("arm64", "aarch64") then
+        target = "aarch64"
     elseif toolchain:is_arch("arm") then
-        target = "arm"
+        target = "armv7"
     end
 
     if toolchain:is_plat("windows") then
@@ -84,7 +84,7 @@ function main(toolchain, suffix)
             toolchain:add("shflags", "-fuse-ld=lld-link" .. suffix)
         end
     elseif toolchain:is_plat("mingw") then
-        target = target .. "-windows-gnu"
+        target = target .. "-w64-windows-gnu"
     end
 
     if target then

